src/core/internal/traits.d

```diff
@@ -220,6 +220,12 @@ template maxAlignment(U...)
     }
 }
 
+template classInstanceAlignment(T)
+if (is(T == class))
+{
+    alias classInstanceAlignment = maxAlignment!(void*, typeof(T.tupleof));
```

Comment:
This doesn't seem to work for fields with explicit alignment:
```
class C { align(16) int x; }
pragma(msg, classInstanceAlignment!C); // 4u on 32-bit
```

Reply:
Neither does the phobos template of the same name.  It's a library limitation.

Reply:
@rainers I can't find a way to get the explicit alignment of class fields:
```D
class C { align(16) int x; }

void main()
{
    auto c = new C();    
    pragma (msg, c.x.alignof);
    pragma (msg, C.x.alignof);
    pragma (msg, C.tupleof[0].alignof);
}
```

```
4LU
4LU
4LU
```

Reply:
The problem (maybe a bug) seems to be that the compiler only looks at the type, not the symbol.